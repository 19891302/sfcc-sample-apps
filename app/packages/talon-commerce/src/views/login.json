{
    "name": "login",
    "label": "Login",
    "themeLayoutType": "login",
    "component": {
        "name": "commerce/loginForm"
    }
}