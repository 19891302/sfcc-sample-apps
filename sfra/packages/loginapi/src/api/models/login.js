'use strict';

const { RESTDataSource } = require('apollo-datasource-rest');

class Login extends RESTDataSource {
    constructor (config) {
        super();
        this.baseURL = config.COMMERCE_BASE_URL;
        this.clientId = config.COMMERCE_APP_API_CLIENT_ID;
    }

    async didReceiveResponse(response) {
        const authToken = response.headers.get('Authorization');
        const body = await response.json();
        return { body, authentication: authToken };
    }

    async getAuthToken(email, password) {
        const body = {
            "type": "credentials"
        };
        const data = await this.post(
            `customers/auth?client_id=${this.clientId}`, body, {
                headers: {
                    "authorization" : `Basic ${btoa(`${email}:${password}`)}==`
                }
            });
        const authToken = data.authentication;
        this.context.token = authToken;
        return data;
    }

}

export const model = (config) => {
    return {
        login: new Login(config)
    }
};