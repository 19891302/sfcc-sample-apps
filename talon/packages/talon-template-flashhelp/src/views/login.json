{
    "name": "login",
    "label": "Login",
    "themeLayoutType": "login",
    "component": {
        "name": "community_flashhelp/loginForm"
    }
}