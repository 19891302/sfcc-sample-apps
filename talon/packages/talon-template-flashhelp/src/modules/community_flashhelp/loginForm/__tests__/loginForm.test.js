import { createElement } from 'lwc';
import Element from 'community_flashhelp/loginForm';
import { getShadowRoot } from "lwc-test-utils";
import getIsUserNamePasswordEnabled from '@salesforce/apex/appLauncher.LoginFormController.getIsUserNamePasswordEnabled';
import login from '@salesforce/apex/applauncher.LoginFormController.loginGetPageRefUrl';
import { Exception } from 'handlebars';

jest.mock('@salesforce/apex/appLauncher.LoginFormController.getIsUserNamePasswordEnabled', () => ({
    default: jest.fn()
}), { virtual: true });

jest.mock('@salesforce/apex/appLauncher.LoginFormController.loginGetPageRefUrl', () => ({
    default: jest.fn()
}), { virtual: true });

const createLoginForm = () => {
    const element = createElement('community_flashhelp-loginForm', { is: Element });
    document.body.appendChild(element);
    return element;
};

const waitForRerender = (callBack) => {
    setTimeout(() => {
        callBack();
    }, 100);
};

describe('community_flashhelp/loginForm', () => {
    beforeEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders correctly when getIsUserNamePasswordEnabled is true', (done) => {
        getIsUserNamePasswordEnabled.mockImplementation(() => Promise.resolve(true));

        const element = createLoginForm();

        return Promise.resolve().then(() => {
            setTimeout(() => {
                expect(element).toMatchSnapshot();
                done();
            }, 100); // wait for rerender
        });
    });

    it('renders correctly when getIsUserNamePasswordEnabled is false', (done) => {
        getIsUserNamePasswordEnabled.mockImplementation(() => Promise.resolve(false));

        const element = createLoginForm();

        return Promise.resolve().then(() => {
            setTimeout(() => {
                expect(element).toMatchSnapshot();
                done();
            }, 100); // wait for rerender
        });
    });

    it('display error message when username password is invalid', (done) => {
        getIsUserNamePasswordEnabled.mockImplementation(() => Promise.resolve(true));
        const errorMessage = "error message";
        login.mockImplementation(() => Promise.resolve(errorMessage));

        global.URL = function (url) {
            throw new Exception(url);
        };

        const element = createLoginForm();

        return Promise.resolve().then(() => {
            const verifyErrorMessage = () => {
                const msg = getShadowRoot(element).querySelector('lightning-formatted-rich-text').value;
                expect(msg).toBe(errorMessage);
                expect(element).toMatchSnapshot();
                done();
            };

            const loginSubmit = () => {
                getShadowRoot(element).querySelector('[data-user]').value = "user";
                getShadowRoot(element).querySelector('[data-password]').value = "password";

                const lightningButton = getShadowRoot(element).querySelector('lightning-button');
                const button = getShadowRoot(lightningButton).querySelector('button');
                button.click();

                waitForRerender(verifyErrorMessage);
            };

            waitForRerender(loginSubmit);
        });
    });
});
