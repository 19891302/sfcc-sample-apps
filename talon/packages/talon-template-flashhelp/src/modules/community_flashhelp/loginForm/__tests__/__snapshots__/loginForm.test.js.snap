// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`community_flashhelp/loginForm display error message when username password is invalid 1`] = `
<community_flashhelp-loginform>
  #shadow-root(open)
    <div
      class="slds-form"
    >
      <fieldset
        class="slds-form-element slds-form_compound"
      >
        <legend
          class="slds-form-element__legend slds-form-element__label"
        >
          Login to the community
        </legend>
        <div
          class="slds-form-element__control"
        >
          <lightning-input
            data-user="true"
          >
            #shadow-root(open)
              <input
                aria-label="Username"
                data-readonly="false"
                data-required="true"
                data-type="text"
                data-value="user"
              />
          </lightning-input>
          <lightning-input
            data-password="true"
          >
            #shadow-root(open)
              <input
                aria-label="Password"
                data-readonly="false"
                data-required="true"
                data-type="password"
                data-value="password"
              />
          </lightning-input>
        </div>
      </fieldset>
      <div
        class="slds-form-element slds-m-top_medium"
      >
        <div
          class="slds-form-element__control"
        >
          <lightning-button
            data-submitbtn="true"
          >
            #shadow-root(open)
              <button>
                Submit
              </button>
          </lightning-button>
        </div>
      </div>
      <div
        class="slds-form-element"
      >
        <div
          class="slds-form-element__control"
        >
          <lightning-formatted-rich-text
            class="slds-text-color_error"
          >
            #shadow-root(open)
              <input
                data-readonly="false"
                data-value="error message"
              />
          </lightning-formatted-rich-text>
        </div>
      </div>
    </div>
</community_flashhelp-loginform>
`;

exports[`community_flashhelp/loginForm renders correctly when getIsUserNamePasswordEnabled is false 1`] = `
<community_flashhelp-loginform>
  #shadow-root(open)
</community_flashhelp-loginform>
`;

exports[`community_flashhelp/loginForm renders correctly when getIsUserNamePasswordEnabled is true 1`] = `
<community_flashhelp-loginform>
  #shadow-root(open)
    <div
      class="slds-form"
    >
      <fieldset
        class="slds-form-element slds-form_compound"
      >
        <legend
          class="slds-form-element__legend slds-form-element__label"
        >
          Login to the community
        </legend>
        <div
          class="slds-form-element__control"
        >
          <lightning-input
            data-user="true"
          >
            #shadow-root(open)
              <input
                aria-label="Username"
                data-readonly="false"
                data-required="true"
                data-type="text"
              />
          </lightning-input>
          <lightning-input
            data-password="true"
          >
            #shadow-root(open)
              <input
                aria-label="Password"
                data-readonly="false"
                data-required="true"
                data-type="password"
              />
          </lightning-input>
        </div>
      </fieldset>
      <div
        class="slds-form-element slds-m-top_medium"
      >
        <div
          class="slds-form-element__control"
        >
          <lightning-button
            data-submitbtn="true"
          >
            #shadow-root(open)
              <button>
                Submit
              </button>
          </lightning-button>
        </div>
      </div>
    </div>
</community_flashhelp-loginform>
`;
